src/control/server/iosrv.go

```diff
@@ -49,6 +50,30 @@ import (
 	"github.com/daos-stack/daos/src/control/security"
 )
 
+// pmixless returns if we are in PMIx-less or PMIx mode.
+func pmixless() bool {
+	if _, ok := os.LookupEnv("PMIX_RANK"); !ok {
+		return true
+	}
+	if _, ok := os.LookupEnv("DAOS_PMIXLESS"); ok {
+		return true
+	}
+	return false
+}
+
+// pmixRank returns the PMIx rank. If PMIx-less, it returns nilRank.
+func pmixRank() rank {
+	s, ok := os.LookupEnv("PMIX_RANK")
+	if !ok {
+		return nilRank
+	}
+	r, err := strconv.ParseUint(s, 0, 32)
+	if err != nil {
+		panic(fmt.Sprintf("PMIX_RANK=%s: %v", s, err))
```

Comment:
This should not panic. Malformed user input is not an exceptional situation. Please change the return signature to include an error value or else just return nil rank.

Reply:
@mjmac, PMIX_RANK is set by orterun internally, not from users. All PMIX_RANK accesses are hacks that will go away once we fully switch to PMIx-less mode.

Reply:
I understand -- but I don't think it's a good practice to ever use panic in production code, even if it's supposed to go away eventually. Why not just return the nil rank?

Reply:
I still think this panic is fine, but I've changed the function to return (rank, error).